R/ocjr-drug-court-SQ780.R

```diff
@@ -1,17 +1,522 @@
+# - Is credit card fraud stuff covered?
+
+# Looking at the excluded charges ----------------------------------------------
+clean_df |>
+  filter(!sq780_drugs, !sq780_props) |>
+  count(count_as_disposed, sort = T) |>
+  print(n = 200)
+
+# Looked at the top 200 most common
+# - "BOGUS CHECK" is on here a few times, that's covered under Section 1541 -- Added to regex above.
+
+# Looking by sq780_statute var -------------------------------------------------
+clean_df |>
+  filter(sq780_drugs | sq780_props) |>
+  select(!c(id, case_type, date_filed, disposition_date, party, uccs_code)) |>
+  View()
+
+clean_df |>
+  select(!c(id, case_type, date_filed, disposition_date, party, uccs_code)) |>
+  filter(sq780_drugs | sq780_props) |>
+  group_by(sq780_statute) |>
+  summarize(
+    n_charges = n()
+  ) |>
+  arrange(desc(n_charges))
+
+# "Unknown / other"s only -- these are currently counted as SQ780 charges, but don't have a
+# specific statute attached yet.
+clean_df |>
+  filter(sq780_drugs | sq780_props) |>
+  filter(sq780_statute == "Unknown / Other") |>
+  count(count_as_disposed, sort = T) |>
+  print(n = 30)
+
+# Summarizing for export =======================================================
+
+# * Total number of SQ780 offenses charged in 2014, 2015, and 2016
+sq780_counts <- clean_df |>
+  filter(sq780_drugs | sq780_props) |> # Filter out non-780 stuff
+  filter(
+    date_filed >= "2014-01-01",
+    date_filed < "2023-01-01",
+  ) |>
+  mutate(year = lubridate::year(date_filed)) |>
+  distinct(id, case_type, date_filed, year) |>
+  count(year, case_type) # You can do this in one step w/out group_by()!
+
+# expanding to non-OSCN counties
+# using statewide formula from simple possession analysis
+sq780_counts |>
+  pivot_wider(names_from = case_type, values_from = n) |>
+  mutate(
+    total = CM + CF,
+    statewide = round(total * 2.3778)
+  ) |>
+  ungroup() |>
+  gt() |>
+  cols_label(contains("CM") ~ "Misdemeanor",
+             contains("CF") ~ "Felony",
+             contains("y") ~ "Year",
+             total ~ "Total (OSCN)",
+             statewide ~ "Statewide Estimate"
+  ) |>
+  tab_header(
+    title = "Simple Possession Drug and Property Crime Convictions",
+    subtitle = "Before and After SQ780 Passing"
+  )
+
+ggplot(sq780_counts, aes(x=year, y=n, group=case_type)) +
+  geom_line(aes(color=case_type), linewidth=0.5, alpha=0.9) +
+  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
+  scale_y_continuous(labels = scales::comma,
+                     limits = c(0, NA)) +
+  theme(legend.position="bottom",
+        legend.title = element_text(size = 10),
+        legend.text = element_text(size = 10)) +
+  theme(
+    plot.title = element_text(size = 14),
+    axis.title = element_text(size = 12),
+    plot.subtitle = element_text(size = 8),
+    plot.caption = element_text(size = 8),
+    axis.text = element_text(size = 10)) +
+  # scale_x_continuous(breaks = 2014:2022) +
```

Comment:
I just took this out because I remade the graph to be in terms of months instead of years at one point, and I just forgot to change it back.